### ISSUE TITLE
Unable to install; failed to install packages to new root

### ISSUE THREAD

**Issue by linuxgirl22** [AnarchyLinux/installer#926]
Hello, and thanks for this great Arch installer. Usually works 100% perfectly. Unfortunately an issue at the moment. 

**Anarchy version**: latest
**Installation method (VM/Host)**: full install to hard drive (+ mate mate-extras)
**System specifications (Processor, GPU, RAM ...)**: Haswell i7, Radeon R9 255, 16 ram

## What actually happened
After updating Anarchy within the installer, and making all choices, including LTS base-devel kernel, nearly everything installs, but it throws an error at the end ... "failed to install packages to new root." Have tried twice, and same result.

**Comment by btd1337** [comment]
I had two errors:

1) `/etc/pacman.conf` malformed file as describe here: https://github.com/AnarchyLinux/installer/issues/923

2) Package `libreoffice-still-pt-BR` not found.

I think that "BR" in uppercase is wrong.

**Comment by linuxgirl22** [comment]
Thanks for linking me, and, yes, it's impossible to install ... I wouldn't know where to begin with manually messing with the iso/install, even though I'm able to do a proper Arch install (with really good notes, lol).

**Comment by condor2** [comment]
Update script and try again
https://github.com/AnarchyLinux/installer/commit/d0af32259ff496ca2087b20251e696241e7b0330

**Comment by erazemkokot** [comment]
@btd1337 The malforming of pacman.conf most likely happens when enabling multilib, where we use a regex. The relevant are [configure_base](https://github.com/AnarchyLinux/installer/blob/bbad46ec5458b21c2e89e47f14ae0bcc422abd29/lib/configure_base.sh#L404), [configure_desktop](https://github.com/AnarchyLinux/installer/blob/bbad46ec5458b21c2e89e47f14ae0bcc422abd29/lib/configure_desktop.sh#L86), [configure_quick](https://github.com/AnarchyLinux/installer/blob/bbad46ec5458b21c2e89e47f14ae0bcc422abd29/lib/configure_quick.sh#L129) and [configure_system](https://github.com/AnarchyLinux/installer/blob/bbad46ec5458b21c2e89e47f14ae0bcc422abd29/lib/configure_system.sh#L200).

I don't know regex and don't have the time to learn it, so if you want, you can check out the files and see which one might be causing the problems.

**Comment by linuxgirl22** [comment]
> @btd1337 The malforming of pacman.conf most likely happens when enabling multilib, where we use a regex. The relevant are [configure_base](https://github.com/AnarchyLinux/installer/blob/bbad46ec5458b21c2e89e47f14ae0bcc422abd29/lib/configure_base.sh#L404), [configure_desktop](https://github.com/AnarchyLinux/installer/blob/bbad46ec5458b21c2e89e47f14ae0bcc422abd29/lib/configure_desktop.sh#L86), [configure_quick](https://github.com/AnarchyLinux/installer/blob/bbad46ec5458b21c2e89e47f14ae0bcc422abd29/lib/configure_quick.sh#L129) and [configure_system](https://github.com/AnarchyLinux/installer/blob/bbad46ec5458b21c2e89e47f14ae0bcc422abd29/lib/configure_system.sh#L200).
> 
> I don't know regex and don't have the time to learn it, so if you want, you can check out the files and see which one might be causing the problems.

Fortunately at this point the Zen installer runs without the issue I opened this thread about, but may I ask if the aim has changed to that people maintain the installer themselves? (that's not possible for me).

**Comment by erazemkokot** [comment]
> but may I ask if the overall aim has changed to that people maintain the installer themselves? (that's not possible for me/health reasons).

No, that was not the aim, but two of us main developers go to university and the third one has a full time job, so we can't spend that much time on Anarchy.

If I have some free time soon, I'll look into the issue but it is not a priority for me, so any outside help is welcome.

**Comment by linuxgirl22** [comment]
Thanks for responding. If it's not a priority that the installer works, there's other things.

**Comment by jmurph22** [comment]
For me it fails on yay. I don't recall an option to not use yay.

**Comment by erazemkokot** [comment]
> For me it fails on yay. I don't recall an option to not use yay.

It depends on which packages you install. Some packages (e.g. icons) are only available from the AUR, so yay gets installed automatically.

If you get an error between pacman and yay, re-download the latest iso and try again, it was already fixed a some time ago and reuploaded.

**Comment by btd1337** [comment]
@condor2 I used the "stable" version 1.0.8, I believe everyone does the same.
In it the error is present. A new version with patches that do not require user intervention should be released.

**Comment by erazemkokot** [comment]
> @condor2 I used the "stable" version 1.0.8, I believe everyone does the same.
> In it the error is present. A new version with patches that do not require user intervention should be released.

Have you updated Anarchy (`anarchy -u`). He is referring to a commit, which gets applied by updating. This is so we don't have to release a new iso for every commit.

**Comment by linuxgirl22** [comment]
I'd used the updated iso, and updated within install, twice, and still it failed. The Zen installer doesn't force use of yay, and succeeds at install. All this needs is the option to choose whether to install yay or not, or to choose a different helper. It was either a recent pacman and/or pamac update needed yay removed, in order to update system correctly. The point of whatever reason people need to use an arch install helper is to not need to correct structural issues such as this. This was working great, before the pacman/pamac updates and yay issues. If it could be fixed, that would be awesome.

**Comment by erazemkokot** [comment]
The thing is, a new 1.0.8 iso was released that fixed the problem, which leads me to believe that you have an older version of 1.0.8. Try re-downloading 1.0.8 and try again, if it doesn't work, I'll look into it when I have the time.

**Comment by condor2** [comment]
Tested now with custmoized xfce, kernel LTS and there is no issue with yay.

You mean zen kernel?

**Comment by linuxgirl22** [comment]
> The thing is, a new 1.0.8 iso was released that fixed the problem, which leads me to believe that you have an older version of 1.0.8. Try re-downloading 1.0.8 and try again, if it doesn't work, I'll look into it when I have the time.

Thanks for responding. I'm pretty sure it was the latest, as I downloaded it freshly a day before beginning this thread, around Nov 7th, but I'll try it again when I have time and post back.

**Comment by btd1337** [comment]
@erazemkokot By default, a release should be working minimally without any errors or requiring user action, as they often do not master these details.

So a commit that makes the installer "work" should not be treated as any commit.

**Comment by erazemkokot** [comment]
@btd1337 As you already know, Arch isn't a distro that holds your hand when installing it. So we too expect our users to be at least aware that they have to update the installer. And I don't agree with your statement when talking about Anarchy or Arch though, Since Arch is a rolling distro we can't know what changes will be implemented and problems, like the recent `yay` vs `pacman` thing can't be predicted. So the only thing we can do is update a release (which we did with 1.0.8). Like you can see above, Condor tested the new release and the issue was not reproducible, so there isn't much we can do. But do know that I understand your frustration and we do try to release our images as bug-free as possible.

**Comment by Feinzer** [comment]
I'd like to report that I am also having this issue. I downloaded the latest iso from releases and just to be sure I also ran anarchy -u before starting the installer.

I personally selected arch-base kernel with clean KDE, multilib, os-prober and no additional software. it throws a "failed to commit transaction (unexpected error)"

I tried editing the anarchy.conf and removed yay but it didn't fix it, I checked the log and no error was shown, it was just downloading and threw the same error.
